lib/src/migrators/module.dart

```
@@ -117,7 +138,86 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
+    }
+  }
+
+  /// If the current stylesheet is the entrypoint, return a string of @forward
+  /// rules to forward all members for which [_shouldForward] returns true.
+  String _getEntrypointForwards() {
+    if (!_scope.isGlobal) {
+      throw StateError('Must be called from root of stylesheet');
+    }
+    if (_upstreamStylesheets.isNotEmpty) return '';
+
+    // Divide all global members from dependencies into sets based on whether
+    // they should be forwarded or not.
+    var shown = <Uri, Set<String>>{};
+    var hidden = <Uri, Set<String>>{};
+    categorizeMember(Uri url, String originalName, String newName) {
+      if (url == _currentUrl) return;
+      if (_shouldForward(originalName)) {
+        shown[url] ??= {};
+        shown[url].add(newName);
+      } else {
+        hidden[url] ??= {};
+        hidden[url].add(newName);
+      }
+    }
+
+    for (var entry in _scope.variables.entries) {
+      if (entry.value is! VariableDeclaration) continue;
+      categorizeMember(entry.value.span.sourceUrl,
+          (entry.value as VariableDeclaration).name, '\$${entry.key}');
+    }
+    for (var entry in _scope.mixins.entries) {
+      categorizeMember(entry.value.span.sourceUrl, entry.value.name, entry.key);
+    }
+    for (var entry in _scope.functions.entries) {
+      categorizeMember(entry.value.span.sourceUrl, entry.value.name, entry.key);
+    }
+
+    // Create a @forward rule for each dependency that has members that should
+    // be forwarded.
+    var forwards = <String>[];
+    for (var url in shown.keys) {
+      var shownCount = shown[url].length;
+      var hiddenCount = hidden[url]?.length ?? 0;
+      var forward = '@forward "${_absoluteUriToDependency(url)}"';
+
+      // When not all members from a dependency should be forwarded, determine
+      // whether to use a `show` clause or a `hide` clause based on the number
+      // of members in each set.
+      if (hiddenCount > 0) {
+        if (shownCount <= hiddenCount) {
```

This is clever, but I think we probably want to decide which style we want to encourage and use that universally. I lean towards defaulting to `hide`.